packages/core/src/parsers/linkProcessor.js

```diff
@@ -23,41 +41,101 @@ const defaultTagLinkMap = {
   }
-
   if (path.isAbsolute(resourcePath) || utils.isUrl(resourcePath) || resourcePath.startsWith('#')) {
     // Do not rewrite.
     return;
   }
 
   const cwd = path.dirname(cwf);
   const fullResourcePath = path.join(cwd, resourcePath);
-  const resourcePathFromRoot = utils.ensurePosix(path.relative(rootPath, fullResourcePath));
+  const resourcePathFromRoot = getResourcePathFromRoot(rootPath, fullResourcePath);
 
+  const linkAttribName = defaultTagLinkMap[node.name];
   node.attribs[linkAttribName] = path.posix.join(baseUrl || '/', resourcePathFromRoot);
 }
-
-function convertMdExtToHtmlExt(node) {
+function convertMdAndMbdExtToHtmlExt(node) {
   if (node.name === 'a' && node.attribs && node.attribs.href) {
+    const hasNoConvert = lodashHas(node.attribs, 'no-convert');
+    if (hasNoConvert) {
+      return;
+    }
     const { href } = node.attribs;
     const hasMdExtension = href.slice(-3) === '.md';
-    const hasNoConvert = lodashHas(node.attribs, 'no-convert');
-    if (hasMdExtension && !hasNoConvert) {
+    if (hasMdExtension) {
       const newHref = `${href.substring(0, href.length - 3)}.html`;
       node.attribs.href = newHref;
+      return;
+    }
+    const hasMbdExtension = href.slice(-4) === '.mbd';
+    if (hasMbdExtension) {
+      const newHref = `${href.substring(0, href.length - 4)}.html`;
+      node.attribs.href = newHref;
+    }
+  }
+}
+
+function validateIntraLink(node, cwf, config) {
+  // checks if the file indicated in the intra link exists
+  let resourcePath = getResourcePath(node, cwf, config.rootPath);
+  if (!resourcePath || utils.isUrl(resourcePath) || resourcePath.startsWith('#')) {
+    return;
+  }
+
+  // remove hash portion (if any) in the resourcePath
+  const resourcePathUrl = url.parse(resourcePath);
+  if (resourcePathUrl.hash) {
+    resourcePath = resourcePathUrl.path;
+  }
+
+  // check if the file is a Page source
+  // since this function is called after convertMdAndMbdExtToHtml,
+  // we can assume that page source mostly has .html extension (unless user indicates no-convert)
+  const hasHtmlExt = resourcePath.slice(-5) === '.html';
+  const hasMdExt = resourcePath.slice(-3) === '.md'; // suppose user indicates no-convert
+  const hasMbdExt = resourcePath.slice(-4) === '.mbd'; // suppose user indicates no-convert
+  if (hasHtmlExt || hasMdExt || hasMbdExt) {
+    // remove ext and '/' first char
+    const resourcePathWithNoExt = fsUtil.removeExtension(resourcePath).substring(1);
+    const isPageSrc = config.addressablePagesSource.includes(resourcePathWithNoExt);
+    // if not pageSrc, it is a raw asset
+    if (isPageSrc) {
+      // Page source file exists
+      return;
     }
   }
+
+  // file is an asset
+
+  // checks if the asset file is an ignored file
+  const fileIgnore = ignore().add(config.ignore);
+  const tempResourcePath = resourcePath.substring(1); // remove the '/' first char in the string
+  if (tempResourcePath !== '' && fileIgnore.ignores(tempResourcePath)) {
```

Comment:
Yup the `!resourcePath` above actually checks for empty string. 

However, I noticed that are cases where the intra link is just a backslash, `/`. Thus, when I remove the backslash from `resourcePath` in line 115, I actually get an empty string. 

I'm thinking of just adding a new condition to the first guard clause to check if the `resourcePath` is only a backslash. I think there's nothing to check for when it's just a backslash, right? Correct me if I'm wrong.

Reply:
hmm good point, a `/` is still a valid link though so we don't want to skip that actually

that reminds me as well, you'll need to take care of cases where the link implicitly points to some `index.html` (e.g. `href="/userGuide/"` and  `href="/userGuide"` both points to `href="/userGuide/index.html"`) for both pages and assets

so at this point if we had a `/` link, it would be `/index.html`

*also note that checking for `href="/userGuide"` is a little more tricky (versus `href="/userGuide/"`) because you'd also have to check for `userGuide` as a raw file 😓 

you can refer to `cli/index.js` which has similar link normalization for the lazy live reload (`-o`) option

Reply:
Alright thanks for raising this up! Will take this into account